hkmc2/shared/src/test/mlscript/handlers/UserThreads.mls

```diff
@@ -0,0 +1,88 @@
+:js
+:effectHandlers
+
+abstract class ThreadEffect with
+  val tasks = []
+  fun start(thread: ThreadEffect -> ()): ()
+  fun fork(thread: ThreadEffect -> ()): ()
+  fun yld(): ThreadEffect
+
+class Lock(locked) with
+  fun lock(th: ThreadEffect) =
+    while locked do
+      th.yld()
+    set locked = true
+  fun unlock() =
+    set locked = false
+
+fun f(x)(h) =
+  print("f " + x)
+  h.yld()
+  print("f " + x)
+  h.yld()
+  print("f " + x)
+
+fun main(h) =
+  print("main start")
+  h.fork(f(0))
+  h.fork(f(1))
+  h.fork(f(2))
+  print("main end")
+
+
+// LIFO
+handle h = ThreadEffect with
+  fun fork(thread)(k) =
+    this.tasks.unshift(thread)
+    k(())
+  fun yld()(k) =
+    this.tasks.push(k)
+    globalThis.eval("runtime").enterHandleBlock(this, () => this.tasks.shift()(this))
```

Comment:
WOW what! Why not just import `Runtime.mls` and then use it? Or add this function to `Predef`.